### PR TITLE
fix(azure): go-import meta header support for Azure DevOps 

### PR DESCRIPTION
**Pull request by hilariocoelho**
## Changes

Fix go-import metadata header for Azure DevOps private repositories

## Context

Azure DevOps private repositories for go modules are not currently working as described in #22660 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/modules/datasource/go/base.ts [hunk 1]
```diff
@@ -172,7 +172,7 @@ export class BaseGoDatasource {
     goModule: string
   ): DataSource | null {
     const importMatch = regEx(
-      `<meta\\s+name="?go-import"?\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)">`
+      `<meta\\s+name="?go-import"?\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)".*?>`
```
Comment (viceice) [inline lib/modules/datasource/go/base.ts:175]:
```suggestion
      `<meta\\s+name="?go-import"?\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)"\\s*>`
```
i think we should only allow whitespace, it should also match your use case.

File: lib/util/common.ts [hunk 2]
```diff
@@ -11,14 +11,19 @@ import { parseUrl } from './url';
...
+  url: string
+): 'gitlab' | 'github' | 'azure' | null {
   const { hostname } = parseUrl(url) ?? {};
   if (hostname === 'github.com' || hostname?.includes('github')) {
     return 'github';
   }
   if (hostname === 'gitlab.com' || hostname?.includes('gitlab')) {
     return 'gitlab';
   }
+  if (hostname === 'dev.azure.com') {
```
Comment (JamieMagee) [inline lib/util/common.ts:24]: `visualstudio.com` is also a valid hostname. `dev.azure.com/myorg` and `myorg.visualstudio.com` are equivalent.

Comment (hilariocoelho) [reply lib/util/common.ts]: added support for `*.visualstudio.com`

**Comment by hilariocoelho** [comment]
@rarkins made some changes according to @viceice and @JamieMagee reviews.

**Review by viceice — DISMISSED** [review]

**Comment by rarkins** [comment]
@hilariocoelho needs coverage completion

**Comment by hilariocoelho** [comment]
> @hilariocoelho needs a test fix. Maybe the regex change broke your test case?

fixed it. it was the regex change suggested previously. it now supports both optional whitespace and `/` at the end of the meta header

**Review by viceice — APPROVED** [review]

File: lib/modules/datasource/go/base.ts [hunk 3]
```diff
@@ -172,7 +172,7 @@ export class BaseGoDatasource {
     goModule: string
   ): DataSource | null {
     const importMatch = regEx(
-      `<meta\\s+name="?go-import"?\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)">`
+      `<meta\\s+name="?go-import"?\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)"\\s*\\/*>`
```
Comment (viceice) [inline lib/modules/datasource/go/base.ts:175]:
```suggestion
      `<meta\\s+name="?go-import"?\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)"\\s*\\/?>`
```
should only have one slash for html conformance

**Review by viceice — DISMISSED** [review]

**Review by JamieMagee — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**